stac_fastapi/mongo/stac_fastapi/mongo/core.py

```diff
@@ -0,0 +1,354 @@
+            },
+        ]
+        collection_list = Collections(
+            collections=serialized_collections or [], links=links
+        )
+        return collection_list
+
+    def get_collection(self, collection_id: str, **kwargs) -> Collection:
+        """Get collection by id."""
+        with self.client.start_session(causal_consistency=True) as session:
+            collection = self.client.stac.stac_collection.find_one(
+                {"id": collection_id}, session=session
+            )
+        base_url = str(kwargs["request"].base_url)
+
+        if not collection:
+            raise NotFoundError(f"{id} not found in database")
+
+        return self.collection_serializer.db_to_stac(collection, base_url)
+
+    def item_collection(
+        self, collection_id: str, limit: int = 10, token: str = None, **kwargs
+    ) -> ItemCollection:
+        """Read an item collection from the database."""
+        links = []
+        response_features = []
+        base_url = str(kwargs["request"].base_url)
+
+        with self.client.start_session() as session:
+            collection_children = self.client.stac.stac_item.find(
+                {"collection": collection_id}, session=session
+            ).sort(
+                [("properties.datetime", pymongo.ASCENDING), ("id", pymongo.ASCENDING)]
+            )
+
+            for item in collection_children:
+                response_features.append(
+                    self.item_serializer.db_to_stac(item, base_url=base_url)
+                )
+
+        context_obj = None
+        if self.extension_is_enabled("ContextExtension"):
+            count = len(response_features)
+            context_obj = {
+                "returned": count if count <= 10 else limit,
+                "limit": limit,
+                "matched": len(response_features) or None,
+            }
+
+        return ItemCollection(
+            type="FeatureCollection",
+            features=response_features,
+            links=links,
+            context=context_obj,
+        )
+
+    def get_item(self, item_id: str, collection_id: str, **kwargs) -> Item:
+        """Get item by item id, collection id."""
+        with self.client.start_session() as session:
+            item = self.client.stac.stac_item.find_one(
+                {"id": item_id, "collection": collection_id}, session=session
+            )
+        base_url = str(kwargs["request"].base_url)
+
+        if not item:
+            raise NotFoundError(f"{item_id} not found in collection {collection_id}")
+
+        return self.item_serializer.db_to_stac(item, base_url)
+
+    def _bbox2poly(self, bbox):
+        poly = [
+            [
+                [float(bbox[0]), float(bbox[1])],
+                [float(bbox[2]), float(bbox[1])],
+                [float(bbox[2]), float(bbox[3])],
+                [float(bbox[0]), float(bbox[3])],
+                [float(bbox[0]), float(bbox[1])],
+            ]
+        ]
+        return poly
```

Comment:
geojson_pydantic (one of stac-fastapi requirement) as a Polygon class that let you create a polygon from bounds
```python
from geojson_pydantic.geometries import Polygon
poly = Polygon.from_bounds(*bbox).dict(exclude_none=True)
```